src/controller/AutoCommissioner.cpp

```diff
@@ -282,9 +282,13 @@ CHIP_ERROR AutoCommissioner::StartCommissioning(DeviceCommissioner * commissione
     }
     mCommissioner            = commissioner;
     mCommissioneeDeviceProxy = proxy;
+    // softAP which transport type is udp also need to setup network
     mNeedsNetworkSetup =
         mCommissioneeDeviceProxy->GetSecureSession().Value()->AsSecureSession()->GetPeerAddress().GetTransportType() ==
-        Transport::Type::kBle;
+            Transport::Type::kBle ||
+        mCommissioneeDeviceProxy->GetSecureSession().Value()->AsSecureSession()->GetPeerAddress().GetTransportType() ==
+            Transport::Type::kUdp;
```

Comment:
This doesn't make sense: how is this different from just setting `mNeedsNetworkSetup` to true?  The right thing to do here would be to pass in enough information from discovery to indicate how we discovered the device, instead of guessing whether it's on the operational network based on the PeerAddress.

Also, this is not the only change needed for SoftAP commissioning using autocommissioner, because it also needs to deliver CommissioningComplete over the operational network, which this PR does not do...

Reply:
@bzbarsky-apple As described in https://github.com/project-chip/connectedhomeip/issues/17487, this PR mainly aims to resolve the issue that the commissionee does not join the operational network. With this fix, the user can finish the softap commissioning process and control the device over operational network.

But the known issue is that the CASE session for CommissioningComplete interaction is over softap link (should be operational network). We will fix this issue in next PR. What do you think?

Reply:
I modified my code. I think if we have the WiFiCredentials or ThreadOperationalDataset, we need to set up the network. Only when **mNeedsNetworkSetup** is true can we enter stage **CommissioningStage::kWiFiNetworkSetup** or **CommissioningStage::kThreadNetworkSetup**, and this is the stage when chip-tool trying trigger joining of operational network at commissionee.